src/PKSim.Core/PKSimStarterRegistrationConvention.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using OSPSuite.Utility.Container;
+using System.Collections.Generic;
+using System.Linq;
+using OSPSuite.Core;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Utility;
+using OSPSuite.Utility.Extensions;
+
+namespace PKSim.Core
+{
+   /// <summary>
+   ///    Register components in the container that are relevant for the PKSimStarter
+   /// </summary>
+   public class PKSimStarterRegistrationConvention : PKSimRegistrationConvention
+   {
+      public override void Process(Type concreteType, IContainer container, LifeStyle lifeStyle)
+      {
+         if (!shouldRegisterType(concreteType))
+            return;
+
+         base.Process(concreteType, container, lifeStyle);
+      }
+
+      private bool shouldRegisterType(Type concreteType)
```

Comment:
Why is this done here and not in the register with the Include exclude. ?

Reply:
Like this => ```scan.Include(shouldRegisterType);```?